shared/update/index.js

```diff
@@ -1,27 +1,31 @@
 import React, {Component} from 'react'
 import {connect} from 'react-redux'
-import UpdateRender from './index.render'
+import UpdateConfirm from './confirm'
+import UpdatePaused from './paused'
 
 class Update extends Component {
   render () {
-    return <UpdateRender {...this.props} />
+    if (this.props.type === 'confirm') {
+      return <UpdateConfirm {...this.props.options} />
+    } else if (this.props.type === 'paused') {
+      return <UpdatePaused {...this.props.options} />
+    } else {
+      return <div/>
+    }
   }
 
   static parseRoute () {
-    return {componentAtTop: {title: 'update'}}
+    return {componentAtTop: {title: 'Update'}}
   }
 }
 
-Update.propTypes = UpdateRender.propTypes
+Update.propTypes = {
+  type: React.PropTypes.oneOf(['confirm', 'paused']).isRequired,
+  options: React.PropTypes.any
+}
 
 export default connect(
-  state => state.update,
-    undefined,
-    (stateProps, dispatchProps, ownProps) => {
-      return {
-        ...stateProps,
-        ...dispatchProps,
-        ...ownProps
-      }
-    }
+  state => {
+    return {updateConfirm: state.updateConfirm, updatePaused: state.updatePaused}
```

Comment:
they are used by the subcomponents? it didn't work unless I have this in... its registering to listen for those state changes? it's possible I don't understand connect though, there is some magic happening thats not totally clear to me